### ISSUE TITLE
Switch to a Swift-friendly Base32 library

### ISSUE THREAD

**Issue by mattrubin** [mattrubin/OneTimePassword#135]
The current dependency for Base32 encoding and decoding exposes an Objective-C API which doesn't bridge nicely to Swift.

**Comment by mattrubin** [comment]
A prerequisite of this is probably converting the last of the legacy tests to Swift, since `OTPTokenSerializationTests` still uses the Objective-C Base32 interface.

**Comment by EmDee** [comment]
@mattrubin Anything to help out here? I noticed that the current `Base32` dependency doesn't play nicely with `Swift 5.3`. 

/cc @nrubin29

**Comment by mattrubin** [comment]
@EmDee Help is always welcome! Can you clarify what issues the current dependency has with Swift 5.3? I'm not seeing any issues with Xcode 12 beta 3.

**Comment by EmDee** [comment]
@mattrubin You're right. It was actually a problem on `Carthage` side (can't tell anymore, if it is due to missing `Swift` interface file, because the "Build for distribution" flag is not set, or because of this linker issue https://github.com/Carthage/Carthage/issues/3019).
